spring-boot/admin/src/main/java/eu/coatrack/admin/model/repository/ProxyRepository.java

```diff
@@ -41,6 +41,8 @@
     @PostFilter("filterObject.owner != null and filterObject.owner.username == authentication.name")
     List<Proxy> findByName(@Param("name") String name);
 
+    Proxy findById(@Param("id") String id);
```

Comment:
Maybe we could use a `@PostFilter("filterObject.id == authentication.name")` to do the filtering based on the calling gateway? However, right now I am not sure if the gateway ID will actually be available in the `authentication` parameter, to be tested.

Reply:
This is now implemented using the ```@PreAuthorize``` annotation.